README.md

```diff
@@ -1,8 +1,55 @@
 # OpenTelemetry Collector Contrib
 This is a repository for OpenTelemetry Collector contributions that are not part of the
-core repository and core distribution of the Collector.
+[core repository](https://github.com/open-telemetry/opentelemetry-collector) and
+core distribution of the Collector. Typically, these contributions are vendor
+specific receivers/exporters and/or components that are either legacy or are only
+useful to relatively small number of users. 
 
 ## Contributing
-
 If you would like to contribute please read [contributing guidelines](https://github.com/open-telemetry/opentelemetry-collector/blob/master/CONTRIBUTING.md)
 before you begin your work.
+
+## Adding New Components
+Before you start please read the [contributing guidelines](https://github.com/open-telemetry/opentelemetry-collector/blob/master/CONTRIBUTING.md).
+
+Any component (receiver, processor, exporter, or extension) needs to implement 
+the interfaces defined on the [core repository](https://github.com/open-telemetry/opentelemetry-collector).
+Familiarize yourself with the interface of the component that you want to write,
+and use existing implementations as reference.
+
+- Create your component under the proper folder and use
+Go standard package naming recommendations.
+- Use a boiler-plate Makefile that just references the one at top level, 
+ie.: `include ../../Makefile.Common` - this allows you to build your component
+with required build configurations for the contrib repo while avoiding building
+the full repo during development.
+- Each component has its own go.mod file. This allows custom builds of the
+collector to take a limited sets of dependencies - so run `go mod` commands as 
+appropriate for your component.
+- Implement the needed interface on your component by importing the appropriate
+component from the core repo.  Follow the pattern of existing components regarding
+config and factory soruce files and tests. 
+- Implement your component as appropriate. Provide end-to-end tests (or mock 
+backend/client as appropriate). Target is to get 80% of code coverage.
```

Comment:
```suggestion
backend/client as appropriate). Target is to get 80% or more of code coverage.
```
I'd prefer to remove the reference to specific percentage once we will implement ratcheting based on coverage percentage (to fail the build if coverage goes down).